src/deprecated.jl

```diff
@@ -447,20 +440,29 @@ end
 
 @deprecate deleterows!(df::DataFrame, inds) delete!(df, inds)
 
-@deprecate by(f::Union{Base.Callable, Pair}, d::AbstractDataFrame, cols::Any;
-   sort::Bool=false, skipmissing::Bool=false) combine(groupby(d, cols, sort=sort,
-                                                              skipmissing=skipmissing), f)
+@deprecate by(f::Base.Callable, d::AbstractDataFrame, cols::Any;
+    sort::Bool=false, skipmissing::Bool=false) combine(f, groupby(d, cols, sort=sort,
+                                                                 skipmissing=skipmissing))
+@deprecate by(f::Pair, d::AbstractDataFrame, cols::Any;
+    sort::Bool=false, skipmissing::Bool=false) combine(first(f) isa ColumnIndex ?
```

Comment:
Can we print different messages depending on whether `first(f) isa ColumnIndex`? Otherwise the deprecation message is going to be totally scary.

Reply:
The problem is that if you have something like:
```
by(df, :a, :a => f1, [:a, :b] => f2)
```
there is a combination of both. But in the single argument case it can be improved

Reply:
Yeah, it's hard to do something perfect in all cases, but if we can make simple cases simple it's much nicer for users.

Reply:
It seems that we will re-introduce `by` so I will wait with this PR till we make that decision :).

Reply:
OK - it seems we drop `by` so I have updated the deprecations